### PR TITLE
feat: allow Neutrino on LND BTC

### PR DESCRIPTION
**Pull request by kilrau**
Closes https://github.com/ExchangeUnion/xud-docker/issues/264

Testing instructions:

* `bash xud.sh -b neutrino --bitcoin-neutrino true`
* test if all xud ctl, status, getinfo, channel opening, swap work as expected

### CODE REVIEW

**Comment by michael1011** [comment]
I just pushed a commit that adds bootnodes for the neutrino client. This PR is ready for review now @kilrau @erkarl @reliveyy

**Comment by kilrau** [comment]
- [ ] On testnet, lndbtc is stuck not unlocking:
```
LND-BTC Info:
┌──────────┬─────────────────────────────────────────────┐
│ Status   │ Cannot read property 'getInfo' of undefined │

testnet > lndbtc-lncli getinfo
[lncli] Wallet is encrypted. Please unlock using 'lncli unlock', or set password using 'lncli create' if this is the first time starting lnd.
testnet > lndbtc-lncli unlock
Input wallet password: 

lnd successfully unlocked!
testnet > lndbtc-lncli getinfo
[lncli] Wallet is encrypted. Please unlock using 'lncli unlock', or set password using 'lncli create' if this is the first time starting lnd.
```
- [ ] on mainnet, lndbtc is unlocking but in a supervisord-restart loop:
```
et, `lncli unlock` to unlock an existing wallet, or `lncli changepassword` to change the password of an existing wallet and unlock it.
Feb 14 23:03:54.000 [notice] Bootstrapped 10%: Finishing handshake with directory server
Feb 14 23:03:54.000 [notice] Bootstrapped 80%: Connecting to the Tor network
Feb 14 23:03:54.000 [notice] Bootstrapped 90%: Establishing a Tor circuit
Feb 14 23:03:56.000 [notice] Bootstrapped 100%: Done
2020-02-14 23:04:27.155 [INF] LNWL: Opened wallet
2020-02-14 23:04:29.160 [INF] LTND: Primary chain is set to: bitcoin
2020-02-14 23:05:29.161 [ERR] LTND: Unable to create chain control: unable to subscribe for zmq block events: dial tcp 192.168.123.42:28332: i/o timeout
2020-02-14 23:05:29.162 [INF] LTND: Shutdown complete
2020-02-14 23:05:29,183 INFO exited: lnd (exit status 1; not expected)
2020-02-14 23:05:30,185 INFO spawned: 'lnd' with pid 89
Waiting for lnd-bitcoin onion address...
Onion address for lnd-bitcoin is 2zxyprwlqexqyynjrzinvd6v5xt7yav6ruvyxqx7xuzhr3vxzqhpieyd.onion
2020-02-14 23:05:30.239 [INF] LTND: Version: 0.8.2-beta commit=v0.8.2-beta, build=production, logging=default
2020-02-14 23:05:30.239 [INF] LTND: Active chain: Bitcoin (network=mainnet)
2020-02-14 23:05:30.240 [INF] CHDB: Checking for schema update: latest_version=11, db_version=11
2020-02-14 23:05:30.249 [INF] RPCS: password gRPC proxy started at 0.0.0.0:8080
2020-02-14 23:05:30.249 [INF] RPCS: password RPC server listening on 0.0.0.0:10009
2020-02-14 23:05:30.250 [INF] LTND: Waiting for wallet encryption password. Use `lncli create` to create a wallet, `lncli unlock` to unlock an existing wallet, or `lncli changepassword` to change the password of an existing wallet and unlock it.
2020-02-14 23:05:31,251 INFO success: lnd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2020-02-14 23:05:57.273 [INF] LNWL: Opened wallet
2020-02-14 23:05:57.365 [INF] LTND: Primary chain is set to: bitcoin
2020-02-14 23:06:57.366 [ERR] LTND: Unable to create chain control: unable to subscribe for zmq block events: dial tcp 192.168.123.42:28332: i/o timeout
2020-02-14 23:06:57.366 [INF] LTND: Shutdown complete
2020-02-14 23:06:57,391 INFO exited: lnd (exit status 1; not expected)
2020-02-14 23:06:58,393 INFO spawned: 'lnd' with pid 111
Waiting for lnd-bitcoin onion address...
Onion address for lnd-bitcoin is 2zxyprwlqexqyynjrzinvd6v5xt7yav6ruvyxqx7xuzhr3vxzqhpieyd.onion
2020-02-14 23:06:58.440 [INF] LTND: Version: 0.8.2-beta commit=v0.8.2-beta, build=production, logging=default
2020-02-14 23:06:58.440 [INF] LTND: Active chain: Bitcoin (network=mainnet)
2020-02-14 23:06:58.440 [INF] CHDB: Checking for schema update: latest_version=11, db_version=11
2020-02-14 23:06:58.450 [INF] RPCS: password gRPC proxy started at 0.0.0.0:8080
2020-02-14 23:06:58.450 [INF] RPCS: password RPC server listening on 0.0.0.0:10009
2020-02-14 23:06:58.450 [INF] LTND: Waiting for wallet encryption password. Use `lncli create` to create a wallet, `lncli unlock` to unlock an existing wallet, or `lncli changepassword` to change the password of an existing wallet and unlock it.
2020-02-14 23:06:59,451 INFO success: lnd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
```
- [ ] missing `#infura = false` config option as per https://github.com/ExchangeUnion/xud-docker/issues/264

**Comment by michael1011** [comment]
I will add the Infura config option in a separate PR.

How did you enable Neutrino? In the config? I just noticed that you also specified the command line argument `--neutrino` in the issue and I didn't implement that one yet.

**Comment by michael1011** [comment]
@kilrau I added the CLI argument `--bitcoin-neutrino`. Please give this PR another try.

`./xud.sh -b neutrino --bitcoin-neutrino true` works fine for me

**Comment by kilrau** [comment]
> I will add the Infura config option in a separate PR.
:+1: -> https://github.com/ExchangeUnion/xud-docker/issues/295

Apart from this, this PR looks good now. Synced to chain, 3 peers. Just waiting for the graph sync to finish.

**Comment by kilrau** [comment]
So far I'm getting all filters from `78.46.126.167`

**Comment by kilrau** [comment]
Can't approve since I opened the PR, but good to go!
- tested `--bitcoin-neutrino true` in combination, works as expected and overwrites config
- tested testnet and mainnet, both lnd's come up fine. Graph sync took ages on mainnet but if finally managed to sync

Good to merge @reliveyy

File: images/utils/launcher/node/bitcoind.py [hunk 1]
```diff
@@ -131,6 +134,9 @@ def status(self):
         if self.external:
             return self.get_external_status()
 
+        if self.neutrino:
+            return "Ready (Conntected to Neutrino)"
```
Comment (kilrau) [inline images/utils/launcher/node/bitcoind.py]: Just saw this typo now. Easy fix!

**Comment by fagunbhavsar** [comment]
Not able to complete the test due to the error:
Mainnet installation is not working - https://github.com/ExchangeUnion/xud-docker/pull/289#issuecomment-587441244

cc: @kilrau @erkarl

**Comment by kilrau** [comment]
This is a different branch.. A test result from one branch doesn't always (actually mostly shouldn't) influence the test result in another branch. Also I suggest that in addition to testing with existing environments, also to test in a completely new machine without cached docker images, e.g. with a VM.

Apart from that I tested this extensively yesterday and already said that this is ready to be merged.

Please go ahead @reliveyy

**Pull request merged**